Compile error in WeylExtraction class with GCC 9

GCC 9 is a bit more fussy with `default(none)` in OpenMP directives and complains as below:

```
../../Source/utils/WeylExtraction.impl.hpp:133:38: error: ‘a_re_part’ not specified in enclosing ‘parallel’
  133 |                         a_re_part[idx] * Y_lm.Real + a_im_part[idx] * Y_lm.Im;
      |                                      ^
../../Source/utils/WeylExtraction.impl.hpp:108:9: error: enclosing ‘parallel’
  108 | #pragma omp parallel for collapse(2) default(none)
```
and similarly for `a_im_part`.

Unfortunately, adding `a_re_part` and `a_im_part` to the `shared(...)` clause breaks it on older compilers.

--- a/Source/utils/WeylExtraction.impl.hpp
+++ b/Source/utils/WeylExtraction.impl.hpp
@@ -106,12 +106,13 @@ WeylExtraction::integrate_surface(int es, int el, int em,
         // of vector) is equal to the first point
 #ifdef _OPENMP
 #if __GNUC__ > 8
-#define OMP_SHARED_CLAUSE shared(es, el, em, integral_re, integral_im, \
-                                 a_re_part, a_im_part)
+#define OPENMP_CONST_SHARED shared(a_re_part, a_im_part)
 #else
-#define OMP_SHARED_CLAUSE shared(es, el, em, integral_re, integral_im)
+#define OPENMP_CONST_SHARED
 #endif
-#pragma omp parallel for collapse(2) default(none) OMP_SHARED_CLAUSE
+#pragma omp parallel for collapse(2) default(none)                             \
+    shared(es, el, em, integral_re, integral_im) OPENMP_CONST_SHARED
+#undef OPENMP_CONST_SHARED
 #endif
         for (int iradius = 0; iradius < m_params.num_extraction_radii;
              ++iradius)

--- a/Source/utils/WeylExtraction.impl.hpp
+++ b/Source/utils/WeylExtraction.impl.hpp
@@ -105,8 +105,13 @@ WeylExtraction::integrate_surface(int es, int el, int em,
         // the function is periodic  and so the last point (implied but not part
         // of vector) is equal to the first point
 #ifdef _OPENMP
-#pragma omp parallel for collapse(2) default(none)                             \
-    shared(es, el, em, integral_re, integral_im)
+#if __GNUC__ > 8
+#define OMP_SHARED_CLAUSE shared(es, el, em, integral_re, integral_im, \
+                                 a_re_part, a_im_part)
+#else
+#define OMP_SHARED_CLAUSE shared(es, el, em, integral_re, integral_im)
+#endif
+#pragma omp parallel for collapse(2) default(none) OMP_SHARED_CLAUSE
 #endif
         for (int iradius = 0; iradius < m_params.num_extraction_radii;
              ++iradius)

